### PR TITLE
Support VecGeom test harness with non-VGDML geometry

### PR DESCRIPTION
**Pull request by sethrj**
@mrguilima Sorry it took a week for me to have this, but this PR should give you the setup needed to build a VecGeom geometry from GDML using the current test harness. The updated `VecGeom.test.cc` has a `TODO` marking on the new test harness which you should be able to easily adapt for your tests.

@stognini In #423 you removed the `GeantSetup::world() -> G4VPhysicalVolume*` member function in favor of pulling the world geometry from some global variable inside `GeantImporter`. At the time nothing else was using the function, but I've reverted that change since now we *do* want to get the volume pointer without having to directly query geant4. I had assumed the change was just a simplification, but please verify that there's nothing wrong with this reverted implementation.

### CODE REVIEW

**Comment by stognini** [comment]
One minor suggestion. This line has been dragged from old iterations for a long time, and it only caught my attention now. I can't directly comment there, so here it is:

in `src/celeritas/ext/GeantSetup.cc`:
```c++
G4UImanager::GetUIpointer()->ApplyCommand("/run/initialize");
```
can be replaced by 
```c++
run_manager_->Initialize();
```
This allows us to drop the `G4UIManager` include as well.

**Review by stognini — APPROVED** [review]

**Comment by sethrj** [comment]
Nice! I'd always thought that line was a bit funny 😄

File: test/celeritas/ext/Vecgeom.test.cc [hunk 1]
```diff
@@ -30,10 +31,17 @@ using namespace celeritas_test;
...
+#    define TEST_IF_CELERITAS_GEANT(name) name
+#else
+#    define TEST_IF_CELERITAS_GEANT(name) DISABLED_##name
+#endif
+
 //---------------------------------------------------------------------------//
 // TESTS
 //---------------------------------------------------------------------------//
-class VecgeomTest : public celeritas_test::GlobalTestBase
+class VecgeomTestBase : virtual public celeritas_test::GlobalTestBase
```
Comment (mrguilima) [inline test/celeritas/ext/Vecgeom.test.cc:44]: Why not a separate file for VecgeomTestBase, like most other `TestBase`s?

File: test/celeritas/GlobalTestBase.hh [hunk 2]
```diff
@@ -139,7 +138,32 @@ class GlobalTestBase : public celeritas_test::Test
...
+ * Reuse geometry across individual tests.
+ *
+ * This is helpful for slow geometry construction or if the geometry has
+ * trouble building/destroying multiple times per execution due to global
+ * variable usage (VecGeom, Geant4).
+ *
+ * The "geometry basename" should be the filename without extension of a
+ * geometry file inside `test/celeritas/data`.
+ */
+class GlobalGeoTestBase : virtual public GlobalTestBase
```
Comment (mrguilima) [inline test/celeritas/GlobalTestBase.hh:154]: Why not creating a separate file for `GlobalGeoTestBase`?

**Review by mrguilima — APPROVED** [review]
Thanks for the changes, I will modify `G4VecgeomConvTest` to use `GlobalGeoTestBase`.

File: test/celeritas/GlobalTestBase.hh (hunk 2 above)
Comment (sethrj) [reply test/celeritas/GlobalTestBase.hh:154]: You got me, I was being lazy 😆

File: test/celeritas/ext/Vecgeom.test.cc (hunk 1 above)
Comment (sethrj) [reply test/celeritas/ext/Vecgeom.test.cc:44]: It's only used in this one file: if we used it in multiple test files we'd need it separately.

**Pull request merged**